dataverse-webapp/src/main/webapp/dataset.xhtml

```diff
@@ -193,7 +193,7 @@
                                                                 and !DatasetPage.canPublishDataset())}">
                                         <!-- Publish Buttons -->
                                         <p:commandLink type="button"
-                                                       styleClass="btn btn-default btn-access #{DatasetPage.locked and !DatasetPage.dataset.latestVersion.inReview ? 'disabled' : ''}"
+                                                       styleClass="btn btn-default btn-access #{(DatasetPage.locked and !DatasetPage.dataset.latestVersion.inReview) or (settingsWrapper.dataCiteInstallation and DatasetPage.dataset.globalIdCreateTime == null) ? 'disabled' : ''}"
```

Comment:
I wonder if we could store that expression in some variable on template?

Reply:
Hmm, but how would you name it though?

Reply:
Something simple `conditionallyDisable` or sth similar.

Reply:
Can you even declare your own variable in xhtml? I don't see such option. On DatasetPage such variable will be unreadable since the method name will tell nothing.

Reply:
I remember that `c:set` should do something similar, but idk if it works precisely that way
https://docs.oracle.com/javaee/7/javaserver-faces-2-2/vdldocs-facelets/toc.htm